src/interpreter/initialization.rs

```diff
@@ -17,8 +17,7 @@ impl<S> Interpreter<S> {
     pub fn init(&mut self, predicate: bool, block_height: u32, tx: Transaction) -> Result<(), InterpreterError> {
         self.tx = tx;
 
-        self.tx
-            .validate_without_signature(self.block_height() as Word, &self.params)?;
+        self.tx.validate_without_signature(block_height as Word, &self.params)?;
```

Comment:
Its odd that we are taking block height as argument here. It should come from the storage, and definitely isn't an arbitrary parameter. Maybe its like this for some outdated compatibility reasons?

We should remove that argument and use the interpreter call, instead.

https://github.com/FuelLabs/fuel-vm/blob/70243e4da9acf1b03f8d4f34ccdf129e3b20c4b0/src/storage/interpreter.rs#L25

`init` is meant to be private. It was moved to public just to speed up some other implementations, but we shouldn't expose these sensible arguments to the user.

Reply:
Feel free to make some commits to this PR if you see a better way.

Reply:
Added an issue to fix it later; we can for now proceed with the fix

https://github.com/FuelLabs/fuel-vm/issues/156

The fix will potentially involve some tests refactor

Reply:
thanks for opening that issue